pkg/clusters/server/server.go

```diff
@@ -0,0 +1,119 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/Kong/kuma/pkg/config/clusters"
+
+	"github.com/Kong/kuma/pkg/core"
+)
+
+var (
+	clusterStatusLog = core.Log.WithName("cluster-status")
+)
+
+type (
+	ClusterStatusServer interface {
+		Start(<-chan struct{}) error
+		StatusHandler(writer http.ResponseWriter)
+	}
```

Comment:
It's not a server anymore. We are kind of mixing HTTP and logic here.

How about this. Here we have
```
type ClustersStatusPoller interface {
  Start(<-chan struct{}) error
  Clusters() Clusters
}
```
(which you add to Runtime)

then move
```
	writer.WriteHeader(http.StatusOK)
	if err := json.NewEncoder(writer).Encode(p.clusters); err != nil {
		clusterStatusLog.Error(err, "failed marshaling response")
	}

```
to `clustersWs()`

also this file should be renamed to `pkg/clusters/status_poller.go`

Reply:
I cann not move the poller to the `pkg/clusters/` because of circular references with `core`. Renaming it to `pkg/clusters/poller/poller.go`

Reply:
And then returning `Clusters` is not really thread safe.

Reply:
do you mean that we can modify objects? change `Clusters []*Cluster` to `Clusters []Cluster` and it should be fine

Reply:
Please check my solutoion.

Reply:
Well, the only hath changes today is only the Active bool field. That is more or less atomic and won't cause issues. However, in the future, this will be a much more dynamic structure. We should be able to add/remove Local CPs on the fly. So it is a bad practice to just share the underlying slice around without Locks.
I have done it the other way around y passing the encoder from the server so, that Clusters can be serialized with a Lock. I believe this is a safer approach.